app/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -8,132 +8,218 @@ import TSnePlotViewRoute from './TSnePlotViewRoute'
 }
 
 // What component each tab type should render, coupled to ExperimentController.java
 const tabTypeComponent = {
-  'results' : TSnePlotViewRoute,
-  'experiment-design' : ExperimentDesignRoute,
-  'supplementary-information' : SupplementaryInformationRoute,
-  'downloads' : DownloadsRoute
+    'results' : TSnePlotViewRoute,
+    'experiment-design' : ExperimentDesignRoute,
+    'supplementary-information' : SupplementaryInformationRoute,
+    'downloads' : DownloadsRoute
 }
 
-const TopRibbon = ({tabs, routeProps}) =>
-  <ul className={`tabs`}>
-    {
-      tabs.map((tab) =>
-        <li title={tab.name} key={tab.type} className={`tabs-title`}>
-          <NavLink to={{pathname:`/${tab.type}`, search: routeProps.location.search, hash: routeProps.location.hash}}
-                   activeClassName={`active`}>
-            {tab.name}
-          </NavLink>
-        </li>
-      )}
-  </ul>
+function shouldRenderTab(tab, commonProps) {
+    const commonRequiredProps = tabCommonValidations.get(tab.type)
+    let shouldRender = true
+
+    if (commonRequiredProps != null) {
+        commonRequiredProps.some(commonProp => {
+            const propValue = commonProps.valueOf(commonProp);
+            if (propValue === 'undefined' || propValue == '' || propValue == null) {
+                console.log(`${tab.type} data missing the required value for the attribute ${commonProp}`)
+                shouldRender = false
+                return false
+            }
+        })
+    }
+
+    const requiredProps = tabValidations.get(tab.type)
+    const tabProps = tab.props
+
+    if (requiredProps != null) {
+        requiredProps.forEach(requiredProp => {
+            // Check if property requires nested object validation
+            if (requiredProp.includes('.')) {
+                const splitProps = requiredProp.split('.')
+                splitProps.forEach(splitProp => {
+                    let table = []
+                    let tableHeader = []
+                    let TableData = []
+
+                    if (isEmptyArray(table)) {
+                        table = tabProps[splitProp] || []
+                        if (table.length == 0) {
+                            shouldRender = false
+                            return false // Early return on failure
+                        }
+                    }
+                    if (isEmptyArray(tableHeader)) {
+                        tableHeader = table[splitProp] || []
+                        if (tableHeader.length == 0) {
+                            shouldRender = false
+                            return false
+                        }
+                    }
+
+                    if (isEmptyArray(TableData)) {
+                        TableData = tableHeader[splitProp]
+                        if (TableData.length == 0) {
+                            shouldRender = false
+                            return false
+                        }
+                    }
+                });
```

Comment:
`;` nearby here, please remove.

Reply:
Sorry,  I didn't commit `TsnePlotViewRouter`  before your review. Now I fixed all format issues as you mentioned here for both files

Reply:
I search for `;` , so no more